zeebe/gateway-mcp/src/main/java/io/camunda/zeebe/gateway/mcp/ConditionalOnMcpGatewayEnabled.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.mcp;
+
+import io.camunda.zeebe.gateway.rest.ConditionalOnRestGatewayEnabled;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.METHOD})
+@Documented
+@ConditionalOnWebApplication
+@ConditionalOnRestGatewayEnabled // TODO not depend on the REST gateway to allow standalone MCP
```

Comment:
Could be OK to rely on this for an initial iteration. If not, we need to refactor a set of configurations (e.g. authentication and potentially security in general) to provide those beans in case REST and/or MCP are enabled.

Reply:
For the very first iteration, this should be OK. We need to separate this for production-readiness, of course. Otherwise, an MCP gateway will always run when REST is enabled.